sklearn/cross_validation.py

```diff
@@ -1820,8 +1821,13 @@ def train_test_split(*arrays, **options):
 
     Returns
     -------
-    splitting : list of arrays, length=2 * len(arrays)
-        List containing train-test split of input array.
+    splitting : length = 2 * len(arrays), 
+        If input is a list, the output will be a list.
+        If numpy array, the output will be a numpy array.
+        If scipy.sparse matrix, the output will be a 
+        scipy.sparse matrix. If input is a pandas dataframe, 
```

Comment:
Please double check. Also, you can add a versionadded 0.16 thing that says this was changed from always casting to a numpy array. Preferably also add an entry to the whatsnew of 0.16 in the API changes section to say that this changed from always casting to a numpy array to preserving the input type.